src/nnue/evaluate_nnue.cpp

```diff
@@ -91,11 +95,20 @@ bool write_parameters(std::ostream& stream, const T& reference) {
 
 
 // Initialize the evaluation function parameters
-static void initialize() {
+static void initialize(NetSize netSize) {
```

Comment:
I still don't quite like that these functions branch a lot based on the `netSize`, but I don't immediately see a good solution. Would probably warrant abstracting the whole network and its metadata into a class, but it could end up being a lot of ugly template metaprogramming this way if the implementations between networks diverge more than how they are now. I guess it's okay to keep it like this.